.github/workflows/BuildAndTest.yaml

```diff
@@ -1,50 +0,0 @@
-name: Build and Test
-
-on: [push, pull_request]
-concurrency:
-  group: build-${{ github.ref }}
-  cancel-in-progress: true
-
-jobs:
-  build:
-    runs-on: ubuntu-latest
-    timeout-minutes: 60
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-
-      - name: Validate Gradle Wrapper
-        uses: gradle/wrapper-validation-action@v1
-
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
-        with:
-          distribution: 'zulu'
-          java-version: 17
-
-      - name: Setup Gradle
-        uses: gradle/gradle-build-action@v2
-
-      - name: Check lint
-        run: ./gradlew lint
-
-      - name: Build all build type and flavor permutations
-        run: ./gradlew assemble
-
-      - name: Run local tests
-        run: ./gradlew test
```

Comment:
Split this into a separate job, so it runs in parallel.

Also makes it clearer in the interface about what checks are passing/failing/in-progress.

<img width="917" alt="Screenshot 2023-10-12 at 12 33 55 AM" src="https://github.com/google/jetpack-camera-app/assets/417355/c239ddbe-5b4f-4090-ae93-7275b81c6737">